dh-grpc-proxy/src/discovery_handler.rs

```diff
@@ -77,7 +77,7 @@ impl DiscoveryHandler for DiscoveryHandlerImpl {
                 if let Err(e) = discovered_devices_sender.send(Ok(response)).await {
```

Comment:
Why is this a loop? Is there a way to use a file watcher to be notified of changes instead of looping?

Reply:
Yes! That's a good point. Think I got stuck on the wasi mentality so yes, we can replace this hole loop with a async watcher, thanks